pythonsdk/src/main/python/vantiqservicesdk.py

```diff
@@ -1,67 +1,131 @@
 import logging
+from typing import Union, Any, Set
 
 from fastapi import FastAPI, APIRouter, WebSocket, WebSocketDisconnect
+from vantiqsdk import Vantiq
+
+CLIENT_CONFIG_MSG = '_setClientConfig'
 
 
 class BaseVantiqServiceConnector:
 
+
     def __init__(self):
-        self.api = FastAPI()
-        self.router = APIRouter()
-        self.router.add_api_route("/healthz", self.__health_check, methods=["GET"])
-        self.router.add_api_websocket_route("/wsock/websocket", self.__websocket_endpoint)
-        self.api.include_router(self.router)
+        self._active_requests: Set = set()
+        self._api = FastAPI()
+        self._router = APIRouter()
+        self._router.add_api_route("/healthz", self._health_check, methods=["GET"])
+        self._router.add_api_route("/status", self._status, methods=["GET"])
+        self._router.add_api_websocket_route("/wsock/websocket", self.__websocket_endpoint)
+        self._api.include_router(self._router)
+        self._client_config: Union[dict, None] = None
+        self._config_set = asyncio.Condition()
 
     @property
-    def service_name(self):
+    def service_name(self) -> str:
         return 'BasePythonService'
 
     @property
-    def app(self):
-        return self.api
+    def app(self) -> FastAPI:
+        return self._api
+
+    async def _get_client_config(self) -> dict:
+        async with self._config_set:
+            if self._client_config is None:
+                await self._config_set.wait()
+            return self._client_config
+
+    async def _get_vantiq_client(self) -> Vantiq:
+        config = await self._get_client_config()
+        client = Vantiq(config['uri'])
+        try:
+            await client.set_access_token(config['accessToken'])
+        except Exception as e:
+            await client.close()
+            raise e
+        return client
 
-    async def __health_check(self):
+    async def _health_check(self) -> str:
         return f"{self.service_name} is healthy"
 
+    def _status(self) -> dict:
+        return {}
+
     async def __websocket_endpoint(self, websocket: WebSocket):
         await websocket.accept()
         try:
+            # Start by asking for our configuration (if we don't have it)
+            if self._client_config is None:
+                config_request = {"requestId": CLIENT_CONFIG_MSG, "isControlRequest": True}
+                await websocket.send_json(config_request, "binary")
+
             while True:
                 # Get the message in bytes and see if it is a ping
                 msg_bytes = await websocket.receive_bytes()
                 if msg_bytes == b'ping':
                     await websocket.send_bytes('pong'.encode("utf-8"))
                     continue
 
-                # Decode the message as JSON
-                request = json.loads(msg_bytes.decode("utf-8"))
-                logging.debug('Request was: %s', request)
+                # Spawn a task to process the message and send the response
```

Comment:
This was why some things were blocking that shouldn't have been.  I think that's all better now ;).